controllers/ControllerAdmin.php

```diff
@@ -116,50 +121,50 @@ public function avatar(){
 		$fileName = $_FILES['avatar']['name'];
 		$extension = ['jpg', 'jpeg', 'gif', 'png'];
 
+		$key = (new Session)->getKey('user');
+
 		if($_FILES['avatar']['error'] > 0){
 			$error =true;
-			echo $this->twig->render('profile.twig', ['error' => $error, 'logged'=> self::LOGGED, 'user'=> $_SESSION['user']['pseudo'], 'value'=> $_SESSION['user']['email']]);
+			echo $this->twig->render('profile.twig', ['error' => $error, 'logged'=> self::LOGGED, 'user'=> $key['pseudo'], 'value'=> $key['email']]);
 			exit();
 		}
 
 		if ($fileSize > $maxSize){
 			$size = true;
-			echo $this->twig->render('profile.twig', ['size' => $size, 'logged'=> self::LOGGED, 'user'=> $_SESSION['user']['pseudo'], 'value'=> $_SESSION['user']['email']]);
+			echo $this->twig->render('profile.twig', ['size' => $size, 'logged'=> self::LOGGED, 'user'=> $key['pseudo'], 'value'=> $key['email']]);
 			exit();
 		}
 
 		$fileExtension = strtolower(substr($fileName, -3));
 
 		if (!in_array($fileExtension, $extension)){
 			$ext =true;
-			echo $this->twig->render('profile.twig', ['ext' => $ext, 'logged'=> self::LOGGED, 'user'=> $_SESSION['user']['pseudo'], 'value'=> $_SESSION['user']['email']]);
+			echo $this->twig->render('profile.twig', ['ext' => $ext, 'logged'=> self::LOGGED, 'user'=> $key['pseudo'], 'value'=> $key['email']]);
 			exit();
 		}
 
 		$uniqName = md5(uniqid(rand(), true));
 		$newNameFile = $uniqName . "." . $fileExtension;
 		$pathFile = "../public/images/avatar/". $newNameFile;
 		$moveFile = move_uploaded_file($tmpName, $pathFile);
+		$oldFile = "../public/images/avatar/" . $key['avatar'];
 
 		if ($moveFile){
-			$oldFile = "../public/images/avatar/" . $_SESSION['user']['avatar'];
 
-			$upload = $this->userManager->updateAvatar($newNameFile);
+			$upload = $this->userManager->updateAvatar($newNameFile, $key);
 
 			if ($upload){
-				$confirm = true;
-				$_SESSION['user']['avatar'] = $newNameFile;
-				echo $this->twig->render('profile.twig', ['confirm' => $confirm, 'logged'=> self::LOGGED, 'user'=> $_SESSION['user']['pseudo'], 'value'=> $_SESSION['user']['email'], 'avatar'=> $_SESSION['user']['avatar']]);
 
-			}
+				$confirm = true;
+				(new Session)->setValueKey('user', 'avatar', $newNameFile);
+				echo $this->twig->render('profile.twig', ['confirm' => $confirm, 'logged'=> self::LOGGED, 'user'=> $key['pseudo'], 'value'=> $key['email'], 'avatar'=> $key['avatar']]);
```

Comment:
Codacy found an issue: [Use of echo language construct is discouraged.](https://app.codacy.com/manual/PatOpen/OC_Blog/pullRequest?prid=5982252)